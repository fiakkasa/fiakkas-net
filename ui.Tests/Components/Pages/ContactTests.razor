@inherits TestContext

@code {
    private readonly IEmailService _emailService = Substitute.For<IEmailService>();

    public ContactTests() : base()
    {
        ComponentFactories.AddStub<DntInputCaptcha>(@<span>@Random.Shared.Next(1000, 99999)</span>);
        Services.AddSingleton(_emailService);
    }

    [Fact]
    public async Task Contact_Should_Render_Form()
    {
        var cut = Render(@<Contact/>);

        await Task.Delay(250);

        cut.FindAll("alert").Should().BeEmpty();
        cut.FindComponent<EditForm>();
        cut.FindComponent<DataAnnotationsValidator>();
        cut.FindComponents<ValidationMessage<string>>().Should().HaveCount(6);
        cut.FindComponents<EagerInputText>().Should().HaveCount(3);
        cut.FindComponents<EagerInputTextArea>().Should().ContainSingle();
        cut.Find("""button[type="button"]""");
        cut.Find("""button[type="reset"]""");
        cut.Find("""button[type="submit"]""");
    }

}