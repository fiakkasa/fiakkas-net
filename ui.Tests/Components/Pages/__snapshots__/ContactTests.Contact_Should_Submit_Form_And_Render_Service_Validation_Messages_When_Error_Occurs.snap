<h1 class="page-header sticky-top d-flex gap-2 justify-content-start align-items-center bg-body"><span>Contact</span>

        </h1>

<div class="card"><div class="card-body"><form class="d-flex flex-column gap-3" blazor:onsubmit="44">

                <div class="small text-danger">Failed to send</div>
                

                <div><label>Email</label>
                    <div class="input-group"><input type="email" tabindex="1" placeholder="you@awesome-email.com" data-kind="email" class="form-control modified valid" value="user@email.com" blazor:oninput="47" />
                        <span class="input-group-text"><i class="bi bi-envelope-at"></i></span></div>
                    </div>

                <div><label>Subject</label>
                    <div class="input-group"><input type="text" tabindex="2" placeholder="What would you like to inquire about?" maxlength="100" data-kind="subject" class="form-control modified valid" value="Subject" blazor:oninput="48" />
                        <span class="input-group-text"><i class="bi bi-pencil"></i></span></div>
                    <div class="d-flex"><div class="flex-fill"></div>
                        <i class="flex-shrink-0 small text-primary">7 / 100</i></div></div>

                <div><label>Message</label>
                    <div class="input-group"><textarea rows="5" tabindex="3" placeholder="Share your thoughts, ideas, or questions..." maxlength="2000" data-kind="message" class="form-control modified valid" value="Message" blazor:oninput="49"></textarea>
                        <span class="input-group-text"><i class="bi bi-journal-text"></i></span></div>
                    <div class="d-flex"><div class="flex-fill"></div>
                        <i class="flex-shrink-0 small text-primary">7 / 2000</i></div></div>

                <div><div class="input-group"><input type="text" tabindex="4" placeholder="Verification Code" maxlength="5" data-kind="verification" class="form-control modified valid" value="12345" blazor:oninput="50" /><span>12345</span><button type="button" class="input-group-text btn btn-primary rounded-0" tabindex="7" blazor:onclick="45"><i class="bi bi-arrow-repeat"></i></button>
                        <span class="input-group-text"><i class="bi-shield-lock"></i></span></div>
                    </div>

                <div class="d-flex justify-content-end me-2 gap-2"><button type="reset" class="btn btn-secondary px-3" tabindex="6" blazor:onclick="46">
                        Reset
                    </button>
                    <button type="submit" class="btn btn-primary px-3 me-4" tabindex="5"><span class="px-3">Submit</span></button></div></form></div></div>
